18 Normalizr/src/pages/student-list/actions/fetchCountryList.ts

```diff
@@ -0,0 +1,28 @@
+import { normalize } from 'normalizr';
+import { actionsEnums } from '../../../common/actionsEnums';
+import { countryApi } from '../../../rest-api/country-api';
+import {arrayOfCountriesSchema} from '../../../schemas/countrySchema';
+import {CountryView} from '../../../model/view/countryView';
+
+export const fetchCountryListRequestStartedAction = () => (dispatcher) => {
+  const promise = countryApi.fetchCountries();dispatcher
+
+  promise.then(
+    data => {
+      const countries = addInvalidCountryEntry(data);
+      const normalizedCountries = normalize(countries, arrayOfCountriesSchema);
+      dispatcher(fetchCountryListRequestCompletedAction(normalizedCountries));
+    }
+  );
+
+  return promise;
+}
+
+const addInvalidCountryEntry = (countries: CountryView[]) => {
+  return [{ id: 0, name: "Select country..." }, ...countries]
```

Comment:
That's true, mmm... but when this value travels to the reducer what would happen? Thinking loud, this case is something that only will happen in the "Editing Student" case that is out from the domain, and this can be rehydrated using a selector?